atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -42,15 +42,24 @@ private const val MIN_SUPPORTED_KGP_VERSION = "1.7.0"
 open class AtomicFUGradlePlugin : Plugin<Project> {
     override fun apply(project: Project) = project.run {
         checkCompatibility()
-        val pluginVersion = rootProject.buildscript.configurations.findByName("classpath")
-            ?.allDependencies?.find { it.name == "atomicfu-gradle-plugin" }?.version
+        // atomicfu version is stored at build time in atomicfu.properties file
+        // located in atomicfu-gradle-plugin resources
+        val pluginVersion = loadPropertyFromResources("atomicfu.properties", "atomicfu.version")
         extensions.add(EXTENSION_NAME, AtomicFUPluginExtension(pluginVersion))
         applyAtomicfuCompilerPlugin()
         configureDependencies()
         configureTasks()
     }
 }
 
+private fun loadPropertyFromResources(propFileName: String, property: String): String {
+    val props = Properties()
+    val inputStream = AtomicFUGradlePlugin::class.java.classLoader!!.getResourceAsStream(propFileName)
+        ?: throw FileNotFoundException("'$propFileName' file was not found in the classpath")
```

Comment:
Suppose I faced this exception in a project that uses AFU, what should I do? File an issue or debug my configuration? It's worth explicitly printing the next step a user has to make.

Reply:
I guess, the file may not be found with older versions of atomicfu-gradle-plugin on the classpath. So, smth like this may help 👀:

"You are applying `kotlinx-atomicfu` plugin of version `0.23.3` or newer, yet we were unable to determine the specific version of the plugin. 
Starting from version `0.23.3` of `kotlinx-atomicfu`, the plugin version is extracted from the `atomicfu.properties` file, which resides within the `atomicfu-gradle-plugin-{version}.jar`. However, this file couldn't be found. 
Please ensure that there are no `atomicfu-gradle-plugin-{version}.jar` with version older than `0.23.3` present on the classpath."